examples/reminders/lib/feature_reminder_list/ui_components/scroll_to_index_widget.dart

```diff
@@ -53,45 +61,97 @@ class _ReminderListScrollViewState extends State<ReminderListScrollView> {
+        physics: const AlwaysScrollableScrollPhysics(),
         controller: _controller,
         keyboardDismissBehavior: ScrollViewKeyboardDismissBehavior.onDrag,
-        slivers: [
-          SliverStickyHeader(
-            header: AppStickyHeader(
-              text: 'Today',
-              color: context.designSystem.colors.secondaryColor,
-            ),
-            sliver: SliverList(
-              delegate: SliverChildBuilderDelegate(
-                (context, i) => AutoScrollTag(
-                  controller: _controller,
-                  index: i,
-                  key: ValueKey(i),
-                  child: AppReminderTile(
-                    reminder: widget.remindersList.list[i],
-                    isLast: i == (widget.remindersList.length - 1),
-                  ),
-                ),
-                childCount: widget.remindersList.list.length,
-              ),
-            ),
-          ),
-          SliverStickyHeader(
-            header: AppStickyHeader(
-              text: 'This month',
-              color: context.designSystem.colors.secondaryColor,
-            ),
-            sliver: SliverList(
-              delegate: SliverChildBuilderDelegate(
-                (context, i) => AppReminderTile(
-                  reminder: ReminderModel.fromIndex(i),
-                  isLast: i == 13,
-                ),
-                childCount: 14,
-              ),
-            ),
-          ),
-        ],
+        useStickyGroupSeparators: true,
+        elements: widget.remindersList.list,
+        groupBy: _groupBy,
+        groupComparator: (value1, value2) =>
+            value1.index.compareTo(value2.index),
+        groupSeparatorBuilder: _groupSeparatorBuilder,
+        indexedItemBuilder: (context, ReminderModel reminder, int index) {
+          final item = widget.remindersList.getItem(index + 1);
+
+          if (item == null &&
+              widget.remindersList.hasNextPage &&
+              widget.remindersList.isNextPageLoading) {
+            return const Padding(
+              padding: EdgeInsets.only(top: 12),
+              child: AppProgressIndicator(),
+            );
+          }
+          return _itemBuilder(index, reminder);
+        },
       );
+
+  Container _itemBuilder(int index, ReminderModel reminder) {
+    return Container(
+      margin: const EdgeInsets.symmetric(
+        horizontal: 10,
+      ),
+      child: AutoScrollTag(
+        controller: _controller,
+        index: index,
+        key: ValueKey(index),
+        child: AppReminderTile(
+          reminder: reminder,
+        ),
+      ),
+    );
+  }
+
+  Widget _groupSeparatorBuilder(Group type) {
+    var title = '';
+    if (type == Group.overdue) {
+      title = _overdue;
+    } else if (type == Group.today) {
+      title = _today;
+    } else if (type == Group.thisMonth) {
+      title = _thisMonth;
+    } else if (type == Group.thisYear) {
```

Comment:
As discussed with @viktorAngelievPrime, please change this year to something more generic

Reply:
Instead of 'This year' the label will be changed to 'In future'